lib/improver/nbhood.py

```diff
@@ -39,69 +39,51 @@
     concatenate_cubes)
 
 
-class BasicNeighbourhoodProcessing(object):
-    """
-    Apply a neigbourhood processing kernel to a thresholded cube.
-
-    When applied to a thresholded probabilistic cube, it acts like a
-    low-pass filter which reduces noisiness in the probabilities.
-
-    The kernel will presently only work with projections in which the
-    x grid point spacing and y grid point spacing are constant over the
-    entire domain, such as the UK national grid projection.
-
-    A maximum kernel radius of 500 grid cells is imposed in order to
-    avoid computational ineffiency and possible memory errors.
+class Utilities(object):
 
     """
+    Utilities for neighbourhood processing.
+    """
 
-    # Max extent of kernel in grid cells.
-    MAX_KERNEL_CELL_RADIUS = 500
+    def __init__(self):
+        pass
 
-    def __init__(self, radii_in_km, lead_times=None, unweighted_mode=False):
+    @staticmethod
+    def cumulate_array(cube):
         """
-        Create a neighbourhood processing plugin that applies a smoothing
-        kernel to points in a cube.
+        Method to calculate the cumulative sum of an array, by first
+        cumulating vertically so that the largest values are in the upper row,
+        and then cumulating horizontally, so that the largest values are in the
+        right hand column.
```

Comment:
The method cumulates along x and y; these are both horizontal coordinates, not vertical. Modify the docstring to reflect this.